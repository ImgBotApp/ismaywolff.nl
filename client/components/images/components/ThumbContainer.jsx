import React, { PropTypes } from 'react'
import { connect } from 'react-redux'
import { getImageEntities, getImageState } from '../selectors'
import Thumb from './Thumb'

export function ThumbContainer({ entities, images, id }) {
  return <Thumb image={entities[id]} isFetching={images.isFetching} />
}

ThumbContainer.defaultProps = {
  entities: {}
}

ThumbContainer.propTypes = {
  entities: PropTypes.object,
  id: PropTypes.string.isRequired,
  images: PropTypes.shape({
    isFetching: PropTypes.bool.isRequired
  }).isRequired
}

const mapStateToProps = state => ({
  entities: getImageEntities(state),
  images: getImageState(state)
})

export default connect(mapStateToProps)(ThumbContainer)
