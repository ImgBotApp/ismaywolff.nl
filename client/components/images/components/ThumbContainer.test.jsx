import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { ThumbContainer } from './ThumbContainer'

describe('<ThumbContainer />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <ThumbContainer
        id={'id'}
        images={{ result: ['id'], isFetching: false }}
      />
    )
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})

