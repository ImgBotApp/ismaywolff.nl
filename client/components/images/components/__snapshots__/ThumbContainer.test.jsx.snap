// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThumbContainer /> renders correctly 1`] = `
<Thumb
  image={Object {}}
  isFetching={false}
/>
`;
